tests/lib/rules/prefer-use-template-ref.js

```diff
@@ -263,6 +263,23 @@ tester.run('prefer-use-template-ref', rule, {
           })
         </script>
       `
+    },
+    {
+      filename: 'multiple-scripts.vue',
+      code: `
+      <template>
+        <div ref="root" :data-a="A" />
+      </template>
+
+      <script>
+      const A = 'foo'
+      </script>
+
+      <script setup>
+      import { useTemplateRef } from 'vue'
+      const root = useTemplateRef('root')
+      </script>
+      `
```

Comment:
Can you please add 2 test cases (one valid, one invalid) with `<script setup>` first and `<script>` second?